nautobot_floor_plan/models.py

```diff
@@ -146,6 +153,128 @@ def validate_no_resizing_with_tiles(self):
                         f"FloorPlan must maintain original size: ({original.x_size}, {original.y_size}), "
                     )
 
+    def generate_labels(self, axis, count):
+        """
+        Generate labels for the specified axis.
+
+        This method creates an instance of FloorPlanLabelGenerator and uses it to generate labels
+        based on the specified axis and count. It will first check for any custom labels defined
+        for the axis and use them if available; otherwise, it will generate default labels.
+        """
+        generator = FloorPlanLabelGenerator(self)
+        return generator.generate_labels(axis, count)
+
+    def reset_seed_for_custom_labels(self):
+        """Reset seed and step values when custom labels are added."""
+        # Only proceed if there are custom labels
+        if not self.custom_labels.exists():
+            return
+
+        changed = False
+        x_has_custom = self.custom_labels.filter(axis="X").exists()
+        y_has_custom = self.custom_labels.filter(axis="Y").exists()
+
+        if x_has_custom and (self.x_origin_seed != 1 or self.x_axis_step != 1):
+            self.x_origin_seed = 1
+            self.x_axis_step = 1
+            changed = True
+
+        if y_has_custom and (self.y_origin_seed != 1 or self.y_axis_step != 1):
+            self.y_origin_seed = 1
+            self.y_axis_step = 1
+            changed = True
+
+        if changed:
+            # Get the current values before updating
```

Comment:
Is this part being unit-tested against? I couldn't tell 100% from the unit-tests